uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/utils/UiAutomatorUtils.java

```diff
@@ -289,6 +289,23 @@ public static void handleButtonClickSafely(@NonNull final String resourceId) {
         }
     }
 
+    /**
+     * Clicks the button element associated to the supplied resource id with a custom timeout.
+     * Do not throw an exception if the button is not found.
+     *
+     * @param resourceId the resource id of the button to click
+     */
+    public static void handleButtonClickSafely(@NonNull final String resourceId, final long existsTimeout) {
```

Comment:
So the tests are not meant to fail with these button clicks not working?

Reply:
Yes, we can attempt a button click without failing if we don't find the button.

Reply:
nit: we can rename to "handleButtonClickIfExists"

Reply:
I like this better, but will include it in march maintenance